### PR TITLE
Add initial apimgt code for apk

### PR DESCRIPTION
**Pull request by RakhithaRR**
## Purpose
This PR moves some of the apimgt logic to APK

Related issue: https://github.com/wso2/apk/issues/121

### CODE REVIEW

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    modelNameSuffix = "DTO"
+    skipOverwrite = true
+    globalProperties = [
+            apis  : "false",
+            models: ""
+    ]
+}
+
+tasks.register("cleanDTOs") {
+    delete "$projectDir/src/gen/java/org/wso2/apk/apimgt/rest/api/admin/v1/dto".toString()
```
Comment (binodmx) [inline common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle:60]: Shall we add `group` and `description` for this custom task?
for ex:-
```
group 'build'
description 'Clean auto generated DTOs'
```

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle [hunk 2]
```diff
@@ -0,0 +1,67 @@
+plugins {
+    id "org.openapi.generator" version "6.2.1"
```
Comment (binodmx) [inline common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle:2]: Shall we use single quotes for those don't have variable names?

File: common-java-libs/build.gradle [hunk 3]
```diff
@@ -0,0 +1,8 @@
+subprojects {
+    apply plugin: 'java'
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
```
[the remark this answers is not in the document]
Comment (RakhithaRR) [reply common-java-libs/build.gradle]: Fixed with [7804ebd](https://github.com/wso2/apk/pull/110/commits/7804ebd19eea6fb6da6fb2d3cfdb7394a003f985)

File: common-java-libs/org.wso2.apk.apimgt.api/build.gradle [hunk 4]
```diff
@@ -0,0 +1,11 @@
+group = 'org.wso2.apk'
+version = '0.1.0-SNAPSHOT'
+description = 'WSO2 APK API Component'
+java.sourceCompatibility = JavaVersion.VERSION_11
```
[the remark this answers is not in the document]
Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.api/build.gradle]: Fixed with [7804ebd](https://github.com/wso2/apk/pull/110/commits/7804ebd19eea6fb6da6fb2d3cfdb7394a003f985)

File: common-java-libs/org.wso2.apk.apimgt.impl/build.gradle [hunk 5]
```diff
@@ -0,0 +1,50 @@
+group = 'org.wso2.apk'
+version = '0.1.0-SNAPSHOT'
+description = 'WSO2 APK Implementation Component'
+java.sourceCompatibility = JavaVersion.VERSION_11
```
[the remark this answers is not in the document]
Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.impl/build.gradle]: Fixed with [7804ebd](https://github.com/wso2/apk/pull/110/commits/7804ebd19eea6fb6da6fb2d3cfdb7394a003f985)

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle (hunk 1 above)
Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle:61]: Fixed with [7804ebd](https://github.com/wso2/apk/pull/110/commits/7804ebd19eea6fb6da6fb2d3cfdb7394a003f985)

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle (hunk 2 above)
Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle]: Fixed with [7804ebd](https://github.com/wso2/apk/pull/110/commits/7804ebd19eea6fb6da6fb2d3cfdb7394a003f985)

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            apis  : "false",
+            models: ""
+    ]
+}
+
+task cleanDTOs {
+    group 'build'
+    description 'Clean auto generated DTOs'
+    delete "$projectDir/src/gen/java/org/wso2/apk/apimgt/rest/api/admin/v1/dto".toString()
+}
```
Comment (binodmx) [inline common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle:77]: Shall we change like this for [Configuration Avoidance](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html)?
```
tasks.register('cleanDTOs', Delete) {
    group 'build'
    description 'Clean auto generated DTOs'
    delete "$projectDir/src/gen/java/org/wso2/apk/apimgt/rest/api/admin/v1/dto"
}
```

File: common-java-libs/settings.gradle [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+include ':org.wso2.apk.apimgt.user'
+include ':org.wso2.apk.apimgt.impl'
+include ':org.wso2.apk.apimgt.init'
+include ':org.wso2.apk.apimgt.rest.api.util'
+include ':org.wso2.apk.apimgt.rest.api.admin.v1.common'
+
+dependencyResolutionManagement {
+    versionCatalogs {
+        libs {
+            from(files("../libs.versions.toml"))
```
Comment (binodmx) [inline common-java-libs/settings.gradle:30]: Shall we add `$rootDir/` to improve readability?
`from(files("$rootDir/../libs.versions.toml"))`

File: common-java-libs/gradle.properties [hunk 8]
```diff
@@ -1,2 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
 group=org.wso2.apk
-version=0.0.1-SNAPSHOT
+version=0.1.0-SNAPSHOT
+openapiGeneratorVersion=6.2.1
```
Comment (binodmx) [inline common-java-libs/gradle.properties:19]: Shall we move this property to `java.gradle`?

File: common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle (hunk 6 above)
Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.rest.api.admin.v1.common/build.gradle]: Fixed with [c8fd2b8](https://github.com/wso2/apk/pull/110/commits/c8fd2b87c31dcae773015688b84f53279bee65a0)

File: common-java-libs/gradle.properties (hunk 8 above)
Comment (RakhithaRR) [reply common-java-libs/gradle.properties]: Fixed with [c8fd2b8](https://github.com/wso2/apk/pull/110/commits/c8fd2b87c31dcae773015688b84f53279bee65a0)

File: common-java-libs/settings.gradle (hunk 7 above)
Comment (RakhithaRR) [reply common-java-libs/settings.gradle]: Fixed with [c8fd2b8](https://github.com/wso2/apk/pull/110/commits/c8fd2b87c31dcae773015688b84f53279bee65a0)

**Review by binodmx — DISMISSED** [review]

**Review by binodmx — APPROVED** [review]

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**